sagemathinc/cocalc
# Add typing to injected props for function components

# Description
Add `useProjectStore` hook to connect a function component to a project store. Seeing how this goes before making more general `useStore` hook.

# Testing Steps
1. Open project new page and trigger the create dialogue in the file listing
1. In the browser console: `smc.redux.getProjectActions([PROJECT-ID]).setState({available_features: {sage: true}});` to see a change in whats available. 
Available options should look like: 

<img width="363" alt="Screen Shot 2020-01-16 at 12 08 29 PM" src="https://user-images.githubusercontent.com/618575/72559119-f52f0b80-3858-11ea-83cf-00b70beeb238.png">


# Relevant Issues

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):

Front end:
